tracpro/polls/tasks.py

```diff
@@ -92,7 +92,9 @@ def fetch_org_updated_runs(org):
     incomplete_responses = Response.get_update_required(org)
 
     if incomplete_responses:
-        runs = client.get_runs(ids=[r.flow_run_id for r in incomplete_responses])
+        runs = []
+        for i in range(0, len(incomplete_responses), 15):
```

Comment:
Yeah, I actually considered that but I realized handling the last list
would be complicated if it does not have 15 members. We will have to first
slice it off as well anyway, so I decided to go with simple slicing.

On Wed, Apr 15, 2015 at 4:12 PM, Evan Wheeler <notifications@github.com>
wrote:

> In tracpro/polls/tasks.py
> <https://github.com/rapidpro/tracpro/pull/6#discussion_r28417320>:
>
> > @@ -92,7 +92,9 @@ def fetch_org_updated_runs(org):
> >      incomplete_responses = Response.get_update_required(org)
> >
> >      if incomplete_responses:
> > -        runs = client.get_runs(ids=[r.flow_run_id for r in incomplete_responses])
> > +        runs = []
> > +        for i in range(0, len(incomplete_responses), 15):
>
> this looks like a great opportunity for itertools!
> maybe something like:
>
> for chunk in itertools.izip(*(iter(incomplete_responses),) * 15):
>     runs += client.get_runs(ids=map(lambda x: x.flow_run_id, chunk)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rapidpro/tracpro/pull/6/files#r28417320>.
>